datacube_stats/statistics.py

```diff
@@ -126,6 +133,59 @@ def compute(self, data):
         return func(data, dim='time')
 
 
+class SquashedStatistic(Statistic):
+    """
+    Compute statistic by squashing data into a :class:`numpy.array` where the
+    data is scaled to [0,1) and stored as `float32`. The data is transposed so that 'variable'
+    and 'time' are the last two dimensions. The `compute` method must reduce on the 'time'
+    dimension.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.original_dims = None
+        self.squashed_dims = None
+        self.output_coords = None
+        self.names = None
+
+    def preprocess(self, data: xarray.Dataset) -> np.array:
+        LOG.debug('Converting Dataset to array')
+        self.names = list(data.data_vars)
+        self.original_dims = list(data.dims)
+        self.squashed_dims = [d for d in data.dims if (d != 'time') and (d != 'variable')] + ['variable', 'time']
+        self.output_coords = {c: v for c, v in data.coords.items() if (c != 'time') and (c != 'source')}
+        data = data.to_array(dim='variable').transpose(*self.squashed_dims).data
+        return data.astype(np.float32)/10000.
+
+    def postprocess(self, data: np.array) -> xarray.Dataset:
+        LOG.debug('Converting array to Dataset')
+        mask = np.isnan(data)
+        tmp = (data*10000.).astype(np.int16)
+        tmp[mask] = -999
+        da = xarray.DataArray(tmp, dims=self.squashed_dims[:-1], coords=self.output_coords)
+        outdims = ['variable'] + [d for d in self.original_dims if d != 'time']
+        ds = da.transpose(*outdims).to_dataset(dim='variable')
+        return ds.rename({i: v for i, v in enumerate(self.names)})
+
+    @abc.abstractmethod
+    def compute(self, data: np.array) -> np.array:
```

Comment:
I've had this discussion with @daleroberts already, but I probably should mention here too that I would prefer `compute` to be a transformation from an `xarray.Dataset` to an `xarray.Dataset`. The `numpy.ndarray` transform, in my opinion, should have a different name. Of course, Dale disagrees, and that's OK. :smile: 

The default implementation of `compute` in my mind goes:
```python
def compute(self, data):
    return self.postprocess(self.compute_raw(self.preprocess(data)))
```
where `compute_raw` is the abstract method to be overriden by subclasses.

Also, we might want to generalize the hard-coded rescaling, nodata filling, and type casting in the future, as this seems very NBAR specific.

Also, the bit that does the magic (in `load_process_save_chunk`) seems to have gone missing among all the rebasing pandemonium!